### ISSUE TITLE
SendMail fails when Message.BccRecipients is explicitly assigned a null value

### ISSUE THREAD

**Issue by pyrox18** [microsoftgraph/msgraph-sdk-dotnet#2230]
**Describe the bug**
When using SendMail.PostAsync to send a Message with the BccRecipients property explicitly assigned a null value, a `Microsoft.Graph.Models.ODataErrors.ODataError` exception with the following message is thrown: `A null value was found for the property named 'bccRecipients', which has the expected type 'Collection(microsoft.graph.recipient)[Nullable=True]'. The expected type 'Collection(microsoft.graph.recipient)[Nullable=True]' cannot be null but it can have null values.`

**To Reproduce**

Minimal code to reproduce error:

```csharp
var body = new SendMailPostRequestBody
{
    Message = new Message
    {
        From = new Recipient
        {
            EmailAddress = new EmailAddress { Address = "someuser@mail.com" }
        },
        ToRecipients = new List<Recipient>
        {
            new Recipient { EmailAddress = new EmailAddress { Address = "someuser@mail.com" }}
        },
        BccRecipients = null,
        Subject = "Test",
        Body = new ItemBody
        {
            ContentType = BodyType.Text,
            Content = "Test"
        }
    }
};

var graphClient = CreateClient(); // helper to create a GraphServiceClient
try
{
    await graphClient
        .Users["someuser@mail.com"]
        .SendMail
        .PostAsync(body);
    Console.WriteLine("Success");
}
catch (Exception ex)
{
    Console.WriteLine("Error: " + ex.Message);
}
```

**Expected behavior**
The Message is supposed to be successfully sent with the BccRecipients property not causing issues with a null value, as the property is nullable.

**Client version**
Microsoft.Graph 5.36.0 running on .NET 8.0.

**Additional context**
Removing the `BccRecipients = null` assignment in the Message (i.e. leaving the `BccRecipients` property as default) will cause the Message to be able to be sent successfully. However, my use case requires me to dynamically create the `BccRecipients` value, so it would be convenient if I was able to just assign a null value directly to `BccRecipients` in my logic. This used to work in v4.x but it broke when I upgraded the SDK to v5.36.0.

**Comment by andrueastman** [comment]
Thanks for raising this @pyrox18 

The SDK will send the `null` values to the API due to the dirty tracking done by the [`backingStore` ](https://github.com/microsoftgraph/msgraph-sdk-dotnet/blob/dev/docs/upgrade-to-v5.md#backing-store). This means that explicitly setting a null value means that it will be sent to the API which isn't allowed by the API. 

Any chance you are able to achieve your desire if you use something like this?
```cs
            var body = new SendMailPostRequestBody
            {
                Message = new Message
                {
                    From = new Recipient
                    {
                        EmailAddress = new EmailAddress { Address = "someuser@mail.com" }
                    },
                    ToRecipients = new List<Recipient>
                    {
                        new Recipient { EmailAddress = new EmailAddress { Address = "someuser@mail.com" }}
                    },
                    Subject = "Test",
                    Body = new ItemBody
                    {
                        ContentType = BodyType.Text,
                        Content = "Test"
                    }
                }
            };
            if (recepeintsToSend != null && !recepeintsToSend.Any())//check if dynamically created collection is null or empy
                body.Message.BccRecipients = recepeintsToSend;
```

**Comment by pyrox18** [comment]
> Thanks for raising this @pyrox18
> 
> The SDK will send the `null` values to the API due to the dirty tracking done by the [`backingStore` ](https://github.com/microsoftgraph/msgraph-sdk-dotnet/blob/dev/docs/upgrade-to-v5.md#backing-store). This means that explicitly setting a null value means that it will be sent to the API which isn't allowed by the API.
> 
> Any chance you are able to achieve your desire if you use something like this?
> 
> ```cs
>             var body = new SendMailPostRequestBody
>             {
>                 Message = new Message
>                 {
>                     From = new Recipient
>                     {
>                         EmailAddress = new EmailAddress { Address = "someuser@mail.com" }
>                     },
>                     ToRecipients = new List<Recipient>
>                     {
>                         new Recipient { EmailAddress = new EmailAddress { Address = "someuser@mail.com" }}
>                     },
>                     Subject = "Test",
>                     Body = new ItemBody
>                     {
>                         ContentType = BodyType.Text,
>                         Content = "Test"
>                     }
>                 }
>             };
>             if (recepeintsToSend != null && !recepeintsToSend.Any())//check if dynamically created collection is null or empy
>                 body.Message.BccRecipients = recepeintsToSend;
> ```

@andrueastman Yes, I am currently using something similar to get what I need. If this is the expected behavior then I guess it might be okay if this issue is closed, but it seems slightly counterintuitive to me as I am just essentially assigning a `null` value to something that was originally already `null`. Would it be possible for the dirty tracking to be modified so that it doesn't count this as a property change?
